src/arch/riscv/insts/vector.cc

```diff
@@ -41,6 +41,74 @@ namespace gem5
 namespace RiscvISA
 {
 
+void roundUnsignedInteger(__uint128_t &result, uint32_t xrm, int gb) {
```

Comment:
roundUnsignedInteger & roundSignedInteger looks exactly the same. And for utility functions, it might be more appropriate to put it in riscv/utility.hh.

Reply:
We seem to have implemented this function as well (haven't synced to GitHub yet) and may need to do some rebase after this PR is merged, but here's our implementation.

```cpp
template<typename T> T
int_rounding(T result, uint8_t xrm, unsigned gb) {
    const uint64_t lsb = 1UL << gb;
    const uint64_t lsb_half = lsb >> 1;
    switch (xrm) {
    case 0 /* RNU */:
        result += lsb_half;
        break;
    case 1 /* RNE */:
        if ((result & lsb_half) &&
            ((result & (lsb_half - 1)) || (result & lsb)))
            result += lsb;
        break;
    case 2 /* RDN */:
        break;
    case 3 /* ROD */:
        if (result & (lsb - 1))
            result |= lsb;
        break;
    default:
        panic("Invalid xrm value %d", (int)xrm);
    }

    return result;
}
```

Reply:
But it's ok to keep it unchanged though, we can't avoid modifying it in a later rebase anyway.